### PR TITLE
feat : Make models jackson compatible for JSONAPI implementation

### LINKED ISSUE #1729: Make models jackson compatible for JSONAPI implementation
**Actual Behaviour**

Currently all models are being serialized/deserialized using gson. For jsonapi we are using `https://github.com/jasminb/jsonapi-converter` which requires https://github.com/FasterXML/jackson-databind.

**Expected Behaviour**

All model should be jackson compatible for jsonapi implementation

**Would you like to work on the issue?**

Yes

**Target Milestone to solve the issue**

[Milestone 7](https://github.com/fossasia/open-event-android/milestone/30)

### PR DESCRIPTION
**Pull request by Shailesh351**
Fixes #1729  

Changes: 
- Transfer from Gson to Jackson

Here Dummy Observable class is needed.  Ref : https://realm.io/docs/java/latest/#jackson-databind

### CODE REVIEW

**Comment by Shailesh351** [comment]
@arp95 @iamareebjamal @shreyasnbhat Please review

File: android/app/build.gradle [hunk 1]
```diff
@@ -36,6 +36,9 @@ android {
         testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
         buildConfigField "String", "GIT_SHA", "\"${gitSha}\""
         buildConfigField "String", "BUILD_TIME", "\"${buildTime}\""
+
+        // Enabling multidex support.
+        multiDexEnabled true
```
Comment (iamareebjamal) [inline android/app/build.gradle:41]: Is the build surpassing 65k methods?

File: android/app/src/main/java/rx/Observable.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package rx;
+
+public class Observable {
+    // Dummy class required for Jackson-Databind support if
+    // RxJava is not a project dependency.
```
Comment (iamareebjamal) [inline android/app/src/main/java/rx/Observable.java:5]: But RxJava is project dependency

**Review by iamareebjamal — CHANGES_REQUESTED** [review]
Some doubts. 

Did you add any field except getters and setters?

File: android/app/build.gradle (hunk 1 above)
Comment (Shailesh351) [reply android/app/build.gradle:41]: Yes, it is because of jackson jar file is large and have more methods

**Comment by Shailesh351** [comment]
@iamareebjamal I have not added new fields. I have planned to add new fields and change Json property values when we switch to JSONAPI

File: android/app/src/main/java/rx/Observable.java (hunk 2 above)
Comment (Shailesh351) [reply android/app/src/main/java/rx/Observable.java:5]: Yes it is but  For realm we have not used Observable. 
https://realm.io/docs/java/latest/#rxjava

**Review by iamareebjamal — APPROVED** [review]

**Review by rohanagarwal94 — APPROVED** [review]

**Review by arp95 — APPROVED** [review]

**Comment by shreyasnbhat** [comment]
@Shailesh351 Sorry for the delay. Looking at this now.

**Pull request merged**